tools/otci/otci/otci.py

```diff
@@ -2535,6 +2535,55 @@ def is_command_supported(self, command: str) -> bool:
 
         return True
 
+    #
+    # Network management utilities
+    #
+    def create_dataset(self,
+                       active_timestamp: Optional[int] = None,
+                       channel: Optional[int] = None,
+                       channel_mask: Optional[int] = None,
+                       extpanid: Optional[str] = None,
+                       mesh_local_prefix: Optional[str] = None,
+                       network_key: Optional[str] = None,
+                       network_name: Optional[str] = None,
+                       panid: Optional[int] = None,
+                       pskc: Optional[str] = None,
+                       security_policy: Optional[tuple] = None,
+                       pending_timestamp: Optional[int] = None) -> bytes:
+        """Creates a new Operational Dataset with given parameters."""
+        self.dataset_clear_buffer()
+        self.dataset_init_buffer()
+        self.dataset_set_buffer(active_timestamp, channel, channel_mask, extpanid, mesh_local_prefix, network_key,
+                                network_name, panid, pskc, security_policy, pending_timestamp)
+        return self.get_dataset_tlvs_bytes()
+
+    def join(self, dataset: bytes) -> None:
+        """Joins to a Thread network with given Active Operational Dataset."""
+        self.set_dataset_bytes('active', dataset)
+        self.ifconfig_up()
+        self.thread_start()
+
+    def leave(self) -> None:
+        """Leaves from the Thread network."""
+        self.thread_stop()
+        self.ifconfig_down()
+
+    def wait_for(self, command: str, expect_line: Optional[Union[str, Pattern, Collection[Any]]], timeout: float = 60):
```

Comment:
nit: Will it be more flexible if you pass in a lambda instead of expect_line?

May consider making a generic version of this and optionally have another function like `wait_for_line` if necessary.

Reply:
After a second thought I think it's fine to keep it as-is for there's no obvious use case requiring a lambda predicate.

Reply:
Based on the offline chat, current implementation of `wait_for` could handle the case such as checking the `state` is `leader` or `router` via calling the function like `wait_for("state", ["leader", "router"])`. This function does not need to be changed